### ISSUE TITLE
"Allow contribution to videos" controls all media publishers, not just videos

### ISSUE THREAD

**Issue by LaurenWags** [brave/brave-browser#13394]
## Description 

Unchecking "Allow contribution to videos" in AC settings keeps any media publisher (Twitter, Reddit, GitHub, Vimeo, Twitch, YouTube) from being added to auto contribute list.

Not sure if this is expected with new greaselion implementation. Prior to greaselion, GitHub was added to the list even if this was unchecked (fixed with https://github.com/brave/brave-browser/issues/10546). So at the very least, this is a regression for GH.

Found while testing https://github.com/brave/brave-browser/issues/12107 and using component version 1.0.40.

## Steps to Reproduce

   1. Clean install 1.19.x (launched with `/Applications/Brave\ Browser\ Dev.app/Contents/MacOS/Brave\ Browser\ Dev --args --enable-logging=stderr --vmodule="*/bat-native-ledger/*"=6,"*/brave_rewards/*"=6,"*/bat-native-ads/*"=6,"*/bat-native-confirmations/*"=6,"*/brave_ads/*"=6,"*/brave_user_model/*"=6 --brave-ads-staging --rewards=staging=true,reconcile-interval=5 --use-dev-goupdater-url`
   2. Click on BAT logo and "Start using Brave Rewards"
   3. Open brave://rewards and open AC settings
   4. Uncheck "Allow contribution to videos"
   5. Restart browser (needed per https://github.com/brave/brave-browser/issues/13391)
   6. Visit a page like github.com/laurenwags for at least 30s
   7. Open brave://rewards
   8. GitHub page is not listed in AC list.
   9. Repeat for other media publisher types

## Actual result:

No media publishers added to AC list

## Expected result:
Non-video media publishers (GitHub, Reddit, Twitter) to be added to AC list, video media publishers (YouTube, Twitch, Vimeo) not added to AC list

## Reproduces how often: 

easily

## Brave version (brave://version info)

Brave | 1.19.76 Chromium: 87.0.4280.101 (Official Build) dev (x86_64)
-- | --
Revision | 9407c80213cda69c2b7abcb4fa8e3f74488f4956-refs/branch-heads/4280@{#1807}
OS | macOS Version 10.15.7 (Build 19H15)

## Version/Channel Information:

- Can you reproduce this issue with the current release? n/a for 1.18.x
- Can you reproduce this issue with the beta channel? yes 1.19.x
- Can you reproduce this issue with the nightly channel? yes 1.20.x

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? n/a
- Does the issue resolve itself when disabling Brave Rewards? n/a
- Is the issue reproducible on the latest version of Chrome? n/a

## Miscellaneous Information:

cc @emerick @kjozwiak @rebron

**Comment by emerick** [comment]
I think our support for media publishers A-C treats all of the media publishers as video-related, so we can definitely take a smarter approach here.

**Comment by Miyayes** [comment]
It seems like it should only disable Vimeo, Twitch, YouTube, which are explicitly video platforms. Does that make sense?

**Comment by LaurenWags** [comment]
sure, but does it only disable the actual playing of a video being added to AC or does it disable all of those platforms entirely from being added to AC? 

Meaning, starting with 1.19.x and Greaselion component 1.0.40 (per this issue: https://github.com/brave/brave-browser/issues/12107) visiting a page such as youtube.com/user/jjdsampson and simply viewing the page for required amount of time (without playing a video) will add it to the AC list.

Feel free to DM me if we should discuss over a call 👍🏻

**Comment by Miyayes** [comment]
> but does it only disable the actual playing of a video being added to AC or does it disable all of those platforms entirely from being added to AC?

Good point @LaurenWags. I would think the former: namely, the actual playing of a video being added to AC. However, question: would a platform page still count in AC? For example, sitting on a YouTube channel's page for 2 hours without actually playing a video? (Just trying to check where the two come apart!)

**Comment by LaurenWags** [comment]
> However, question: would a platform page still count in AC? For example, sitting on a YouTube channel's page for 2 hours without actually playing a video?

Yes, starting with 1.19.x with GL component version 1.0.40 or higher as per https://github.com/brave/brave-browser/issues/12107.

**Comment by Miyayes** [comment]
Thanks. If I had to choose between these options (and there is no perfect third way), I would remove those video publishers completely from A-C. 

I can imagine scenario (A) occurring much more than (B), and (A) arousing much more anger:

* Scenario (A): "I disallowed 'contributions to videos', yet I'm still seeing contributions to all these YouTubers! Why are you not respecting my choices?" 

* Scenario (B): "I disallowed 'contributions to videos', but I still want to donate to my YouTubers when I'm sitting on their pages but not watching their videos. Why won't A-C let me do that?!"

Intuitively, I think users who really wanted to support their YouTubers despite disabling "allow contributions to videos" would just do it through tipping. Moreover, (A) results in _unexpected loss of BAT_ (which is the most aggravating situation for users), whereas (B) would never lead to unexpected loss. (B) at most would lead to unexpected _keeping of_ BAT! 

So, _remove those video platforms completely from A-C when 'allow contribution to videos' is disabled_ would be my decision. 

Open for feedback/thoughts on above reasoning :).
